test/sparse/sparse.jl

```diff
@@ -945,886 +1004,951 @@ function test_getindex_algs{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector,
-let X = sparse([1 -1; -1 1])
-    for T in (Symmetric, Hermitian)
-        Y = T(copy(X))
-        _Y = similar(Y)
-        copy!(_Y, Y)
-        @test _Y == Y
-
-        W = T(copy(X), :L)
-        copy!(W, Y)
-        @test W.data == Y.data
-        @test W.uplo != Y.uplo
-
-        W[1,1] = 4
-        @test W == T(sparse([4 -1; -1 1]))
-        @test_throws ArgumentError (W[1,2] = 2)
-
-        @test Y + I == T(sparse([2 -1; -1 2]))
-        @test Y - I == T(sparse([0 -1; -1 0]))
-        @test Y * I == Y
+@testset "Issue #18705" begin
+    @test isa(sin.(spdiagm(1.0:5.0)), SparseMatrixCSC)
+end
 
-        @test Y + 1 == T(sparse([2 0; 0 2]))
-        @test Y - 1 == T(sparse([0 -2; -2 0]))
-        @test Y * 2 == T(sparse([2 -2; -2 2]))
-        @test Y / 1 == Y
+@testset "Issue #19225" begin
+    let X = sparse([1 -1; -1 1])
+        for T in (Symmetric, Hermitian)
+            Y = T(copy(X))
+            _Y = similar(Y)
+            copy!(_Y, Y)
+            @test _Y == Y
+
+            W = T(copy(X), :L)
+            copy!(W, Y)
+            @test W.data == Y.data
+            @test W.uplo != Y.uplo
+
+            W[1,1] = 4
+            @test W == T(sparse([4 -1; -1 1]))
+            @test_throws ArgumentError (W[1,2] = 2)
+
+            @test Y + I == T(sparse([2 -1; -1 2]))
+            @test Y - I == T(sparse([0 -1; -1 0]))
+            @test Y * I == Y
+
+            @test Y + 1 == T(sparse([2 0; 0 2]))
+            @test Y - 1 == T(sparse([0 -2; -2 0]))
+            @test Y * 2 == T(sparse([2 -2; -2 2]))
+            @test Y / 1 == Y
+        end
     end
 end
 
-# 19304
-@inferred hcat(sparse(rand(2,1)), eye(2,2))
+@testset "Issue #19304" begin
+    @inferred hcat(sparse(rand(2,1)), eye(2,2))
+end
 
 # Test that broadcast[!](f, [C::SparseMatrixCSC], A::SparseMatrixCSC, B::SparseMatrixCSC)
 # returns the correct (densely populated) result when f(zero(eltype(A)), zero(eltype(B))) != 0
-let
-    N = 5
-    sparsesqrmat = sprand(N, N, 0.5)
-    sparsesqrmat2 = sprand(N, N, 0.5)
-    sparserowmat = sprand(1, N, 0.5)
-    sparsecolmat = sprand(N, 1, 0.5)
-    sparse1x1matz = spzeros(1, 1)
-    sparse1x1mato = spones(sparse1x1matz)
-    zeroscourge = (x, y) -> x + y + 1
-    # test case where the matrices have the same shape and no singleton dimensions
-    @test broadcast(zeroscourge, sparsesqrmat, sparsesqrmat2) ==
-            broadcast(zeroscourge, Matrix(sparsesqrmat), Matrix(sparsesqrmat2))
-    # test combinations where either or both matrices have one or more singleton dimensions
-    sparsemats = (sparsesqrmat, sparserowmat, sparsecolmat, sparse1x1matz, sparse1x1mato)
-    densemats = map(Matrix, sparsemats)
-    for (sparseA, denseA) in zip(sparsemats, densemats)
-        for (sparseB, denseB) in zip(sparsemats, densemats)
-            @test broadcast(zeroscourge, sparseA, sparseB) ==
-                    broadcast(zeroscourge, denseA, denseB)
+@testset "Test broadcast[!] returns the correct (densely populated) result" begin
```

Comment:
> this is where it gets confusing, that is a really long name if you include all the type info and specific conditions.

Agreed, guidelines for testset descriptions would help. This testset description seems potentially confusing when read in isolation though ("Check that sparse broadcast[!] produces a dense result? But shouldn't the result be sparse?"), and a small modification could largely mitigate that confusion potential. Best!